vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java

```diff
@@ -471,4 +512,22 @@ enum SequenceAnalysisType {
       this.label = label;
     }
   }
+
+  public static class SpreadsheetValidationException extends ApplicationException {
+
+    @Serial
+    private static final long serialVersionUID = 3532483874741500810L;
+
+    private final int invalidRow;
+
+    SpreadsheetValidationException(String message, int invalidRow) {
+      super(message);
+      this.invalidRow = invalidRow;
+    }
+
+    public int getInvalidRow() {
+      return invalidRow;
+    }
+  }
```

Comment:
why not simply use the row index?

Reply:
that is the row index

Reply:
instead of the whole Exception thing. You could have a Result<..., Integer>

Reply:
The Exception has been removed, it is now part of an Enum including row index, type of Invalidation and optional message